### ISSUE TITLE
amazonses.com not fixed

### ISSUE THREAD

**Issue by karamble** [schweikert/postgrey#48]
This issue is still present. Amazon Cloud ALLWAYS runs into greylist! BAD

Alot of providers use their smtp servers, twitter, dropbox, bitcoinde, coinbase ..... as seen in the amazon outage them are a lot and the list can be continued. Best would be to azzept amazonses probably.

How you find out which IPs are used for mail delivery trough the amazon cloud:
dig TXT amazonses.com +short| grep "v=spf1"

Hostnames looks like:
a11-22.smtp-out.amazonses.com[54.240.11.22]
a11-56.smtp-out.amazonses.com[54.240.11.56]
a11-52.smtp-out.amazonses.com[54.240.11.52]
a11-35.smtp-out.amazonses.com[54.240.11.35]

You are missing a leading wildcard on your greylist to make amazonses work!

The problem is, amazonses tries to deliver mail with a different host/ip each time. So the greylist idea never works in this case. We are missing a leading wildcard in upgfront of your f9fb0ec commit.

add amazonses like this:
/^.*\.amazonses\.com/

**Comment by starrychloe** [comment]
Wow I was just going to enter an issue for Amazon SES! I couldn't find anything about AmazonSES in the archive search (http://lists.ee.ethz.ch/wws/arc/postgrey) and I didn't want to subscribe, but I thought I would bring it to your attention. I think the problem is with Amazon because it keeps trying with a random email and server, and I think that violates RFC, but I'm not sure. So I'll just leave this info here and maybe there is a creative solution.

```
[root@kizbit ~]# cat /var/log/maillog | /usr/sbin/postgreyreport --delay=300 | grep amazonses | grep starrychloe
    54.240.8.80 a8-80.smtp-out.amazonses.com 0100015ac421be2e-13d5c0a8-a6df-4a3e-8a73-105b19faff03-#@amazonses.com starrychloe@oliveyou.net
    54.240.8.91 a8-91.smtp-out.amazonses.com 0100015ac421be2e-13d5c0a8-a6df-4a3e-8a73-105b19faff03-#@amazonses.com starrychloe@oliveyou.net
    54.240.8.99 a8-99.smtp-out.amazonses.com 0100015ac421be2e-13d5c0a8-a6df-4a3e-8a73-105b19faff03-#@amazonses.com starrychloe@oliveyou.net
   54.240.9.108 a9-108.smtp-out.amazonses.com 0100015acd65a158-4952ff79-f250-49f9-a37d-10fac2f79e53-#@amazonses.com starrychloe@oliveyou.net
   54.240.10.18 a10-18.smtp-out.amazonses.com 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com starrychloe@oliveyou.net
   54.240.10.20 a10-20.smtp-out.amazonses.com 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com starrychloe@oliveyou.net
   54.240.10.33 a10-33.smtp-out.amazonses.com 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com starrychloe@oliveyou.net
   54.240.10.95 a10-95.smtp-out.amazonses.com 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com starrychloe@oliveyou.net
   54.240.10.96 a10-96.smtp-out.amazonses.com 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com starrychloe@oliveyou.net
  54.240.10.161 a10-161.smtp-out.amazonses.com 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com starrychloe@oliveyou.net
  54.240.10.181 a10-181.smtp-out.amazonses.com 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com starrychloe@oliveyou.net
    54.240.11.9 a11-9.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.12 a11-12.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.18 a11-18.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.23 a11-23.smtp-out.amazonses.com 0100015add540a13-1fde9c1b-979f-4ce6-#-e8acf51f2559-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.23 a11-23.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.24 a11-24.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.25 a11-25.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.26 a11-26.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.38 a11-38.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.39 a11-39.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.40 a11-40.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.40 a11-40.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.41 a11-41.smtp-out.amazonses.com 0100015add540a13-1fde9c1b-979f-4ce6-#-e8acf51f2559-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.43 a11-43.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.43 a11-43.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.50 a11-50.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.56 a11-56.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.72 a11-72.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.74 a11-74.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
   54.240.11.74 a11-74.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
  54.240.11.106 a11-106.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
  54.240.11.107 a11-107.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
  54.240.11.133 a11-133.smtp-out.amazonses.com 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com starrychloe@oliveyou.net
  54.240.11.133 a11-133.smtp-out.amazonses.com 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com starrychloe@oliveyou.net
```

And some counts:
```
[root@kizbit ~]# cat /var/log/maillog | postgreyreport | awk '{print $3}' | sort | uniq -c | sort -nr | head -n20
     12 0100015ad3a1bd19-3a06e9aa-d72c-4c2d-af61-0aa8f8172e9c-#@amazonses.com
...
     10 0100015ad7e29a84-f4bb8c77-ff56-476c-aefd-c07989ea672f-#@amazonses.com
      7 0100015acfbd9001-5216a5fc-ffd8-4c42-9d22-5f66b8462154-#@amazonses.com
      3 0100015ac421be2e-13d5c0a8-a6df-4a3e-8a73-105b19faff03-#@amazonses.com
...
      2 0100015acdc67f87-46fc1c83-d52e-4f26-ba53-36c391e95fa1-#@amazonses.com
...
 ```

**Comment by hostingnuggets** [comment]
Thanks @karamble for opening this issue. I hope @schweikert will soon notice that something is fishy here and that his whitelist needs to be amended.

**Comment by schweikert** [comment]
What version of postgrey are you using? amazonses.com is already whitelisted, but it looks like you have encountered the problem described in #32, fixed in version 1.37.
